File: app/src/main/java/com/esafirm/androidplayground/common/BaseAct.java
package com.esafirm.androidplayground.common;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.esafirm.androidplayground.utils.Logger;

public class BaseAct extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Logger.clear();
    }
}

File: app/src/main/java/com/esafirm/androidplayground/common/BaseAct.kt
package com.esafirm.androidplayground.common

import android.annotation.SuppressLint
import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.esafirm.androidplayground.utils.ContextProvider

import com.esafirm.androidplayground.utils.Logger

@SuppressLint("Registered")
open class BaseAct : AppCompatActivity() , ContextProvider{

    override val requiredContext: Context
        get() = this

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Logger.clear()
    }
}
